media-proxy/src/mesh/conn_rdma.cc

```diff
@@ -205,76 +236,186 @@ Result Rdma::configure(context::Context& ctx, const mcm_conn_param& request,
  */
 Result Rdma::on_establish(context::Context& ctx)
 {
+    // 1) Already initialized?
     if (init) {
         log::error("RDMA device is already initialized")("state", "initialized");
         set_state(ctx, State::active);
         return Result::error_already_initialized;
     }
 
+    // 2) Unblock buffer threads
     init_buf_available();
 
-    int ret;
+    int    ret;
     Result res;
 
+    // 3) Initialize the RDMA device if needed
     if (!m_dev_handle) {
+        m_dev_handle = (libfabric_ctx*)calloc(1, sizeof(libfabric_ctx));
+        if (!m_dev_handle) {
+            log::error("Failed to allocate RDMA context")("error", strerror(errno));
+            return Result::error_out_of_memory;
+        }
+        _kind = kind();
+        m_dev_handle->kind        = (_kind == Kind::receiver ? KIND_RECEIVER : KIND_TRANSMITTER);
```

Comment:
Maybe use a prefix to make the enum constants names less general?
```suggestion
        m_dev_handle->kind        = (_kind == Kind::receiver ? FI_KIND_RECEIVER : FI_KIND_TRANSMITTER);
```

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/400/commits/b33649ac18bdc98cceb5669873c24ae363828c91